plugins/redis.js

```diff
@@ -37,24 +37,31 @@ exports.load_redis_ini = function () {
 exports.init_redis_connection = function (next, server) {
     var plugin = this;
 
-    // this is the server-wide redis, shared by plugins that don't
-    // set a specific db ID.
-    if (server.notes.redis && server.notes.redis.ping()) {
-        plugin.loginfo('already connected');
-        return next(); // connection is good
-    }
-
     var calledNext = false;
-    function callNext () {
+    function nextOnce () {
         if (calledNext) return;
         calledNext = true;
         next();
     }
 
-    var opts = plugin.redisCfg.opts;
-    opts.host = plugin.redisCfg.server.host;
-    opts.port = plugin.redisCfg.server.port;
-    server.notes.redis = plugin.get_redis_client(opts, callNext);
+    // this is the server-wide redis, shared by plugins that don't
+    // set a specific db ID.
+    if (server.notes.redis) {
+        server.notes.redis.ping(function (err, res) {
```

Comment:
That wouldn't do the same thing. The sticky part is doing The Right Thing under several circumstances:

1. redis works
2. no redis at all (Appveyor, local devs running `./run_tests`)
3. a redis that recently stopped responding (#1696)

The failure behavior is different for each. For tests, we want the client to give up quickly and scurry along, whereas in production we want it to retry and gracefully degrade after some period.

I did add a check in one of the tests that uses redis.connected and when I get around to #1696, I'll do some empirical testing to see how far I'm willing to trust that property (vs getting a response back from the server).